### PR TITLE
AMDPARs with "paragraph xxx of § YYY"

### PR DESCRIPTION
**Pull request by cmc333333**
For #31. This construction is very prominent in FEC regulations, but likely appears more widely.

Also some flake8 updates, as my text editor tells me so.

### CODE REVIEW

File: regparser/grammar/amdpar.py [hunk 1]
```diff
@@ -206,23 +209,34 @@ def _paren_join(elements):
...
     unified.marker_part_section
     + unified.depth1_p
     + Optional(intro_text_marker)
-    ).setParseAction(lambda m: tokens.Paragraph([m.part, None,
-        m.section, m.p1, m.p2, m.p3, m.p4, m.plaintext_p5, m.plaintext_p6],
+    ).setParseAction(lambda m: tokens.Paragraph([
+        m.part, None, m.section, m.p1, m.p2, m.p3, m.p4, m.plaintext_p5,
+        m.plaintext_p6],
         field=(tokens.Paragraph.TEXT_FIELD if m[-1] == 'text' else None)))
+single_par_section = (
```
Comment (khandelwal) [inline regparser/grammar/amdpar.py]: This is going to sound typical from me. But I think we need to get to the point where we are putting examples of what the grammar parses in a comment. The tests help with this, but it's another file to go look at. 

Thoughts?

Comment (willbarton) [inline regparser/grammar/amdpar.py]: I absolutely agree! The tests are usually fairly descriptive, but they're all the way in another file.

**Comment by khandelwal** [comment]
Also the amdpar.py file is now 440 lines long. I can only imagine that it's going to get longer. Time to think about how to break it up? Would that help with documentation?

File: regparser/grammar/amdpar.py (hunk 1 above)
Comment (cmc333333) [inline regparser/grammar/amdpar.py]: :+1:

**Comment by cmc333333** [comment]
Updated!

> Time to think about how to break it up? Would that help with documentation?

Indeed, many of the grammars are quite large and unhelpfully flat. They're a heavily trodden, area, though. Let's circle back to splitting them out and make sure we're on the same page with @willbarton, @grapesmoker, et. al. ?

**Pull request merged**